### ISSUE TITLE
CSS variables flagged as Parse Error

### ISSUE THREAD

**Issue by morganwdavis** [validator/validator#644]
Valid CSS variables for user agent are flagged as CSS Parse Error.

**Comment by morganwdavis** [comment]
I tried it with both sites and I get the same results (more errors on checker.html5.org).  Here's the site I tested it with: https://www.jeromes.com.  It has minified, inlined CSS, with fairly complex calc()s for computing varying flex layouts with gutters.

Rather than trying to interpret that, here's a tiny test that fails the same way:
```
#test {
  --x: 5;
  --y: 3;
  --z: calc(100% / var(--x) - var(--y));
}
```
Regardless of the validator's report, Chrome is able to interpret this properly.

**Comment by sideshowbarker** [comment]
OK, thanks, yeah looking at the following:

https://validator.w3.org/nu/?showsource=yes&doc=data%3Atext%2Fhtml%3Bcharset%3Dutf-8%2C%253C%2521DOCTYPE%2520html%253E%250D%250A%253Chtml%2520lang%253D%2522%2522%253E%250D%250A%253Chead%253E%250D%250A%253Ctitle%253ETest%253C%252Ftitle%253E%250D%250A%253Cstyle%253E%250D%250A%2523test%2520%257B%250D%250A%2520%2520--x%253A%25205%253B%250D%250A%2520%2520--y%253A%25203%253B%250D%250A%2520%2520--z%253A%2520calc%2528100%2525%2520%252F%2520var%2528--x%2529%2520-%2520var%2528--y%2529%2529%253B%250D%250A%257D%250D%250A%253C%252Fstyle%253E

…I see the specific problem seems to be with handling of `var` within `calc`. I’ll investigate.

**Comment by sideshowbarker** [comment]
I’ve “fixed” the behavior for the case in the snippet in https://github.com/validator/validator/issues/644#issuecomment-383254214. See https://goo.gl/6XChFZ

However, I think there are other cases that can still fail, and I’m not super-confident the change I made doesn’t introduce any regressions or otherwise have unintended side effects.

The thing is, the current CSS variables “support” in the checker is a kind of crude hack that’s just meant as an interim workaround until we manage to get it implemented in the right way. I’m still a bit surprised I managed to get it working at all…

But anyway, if it works at least for the documents you’re checking, then I guess we could go ahead and close this issue. So lemme know.

And then If you later run into a case later that it doesn’t work for, then you could open up a new issue.

**Comment by sideshowbarker** [comment]
OK — I’ll go ahead and close this.

Thanks much for having taken time to report the problem. I’m glad I was able to actually fix it for your cases with a relatively minimal amount of re-wrestling with the code.

Anyway, if/when you do run into any new cases that it chokes on, please open new issues.
